### ISSUE TITLE
Name conflict - Swift - senderId and senderDisplayName

### ISSUE THREAD

**Issue by tomoueda** [jessesquires/JSQMessagesViewController#918]
I just updated to Xcode 6.3. I'm developing in swift. This wasn't an issue before, but after upgrading to the new Xcode the compiler began complaining that my messageViewController has a obj-c getter called senderId(),  so I can't declare senderId() inside JSQMessageCollectionViewDataSource. Is there a quick fix for this?

**Comment by francisjervis** [comment]
I am seeing this too, same for senderDisplayName.

**Comment by ZhuchoG** [comment]
me to :(

**Comment by Fjohn666** [comment]
I got the same problem her :(

**Comment by jessesquires** [comment]
@Fjohn666 's solution is correct.

You simply need to set this properties in `viewDidLoad`

**Comment by francisjervis** [comment]
Sorry, could you clarify that a little? I removed the 2 functions & tried setting the properties as described, but now get a "does not conform to protocol JSQ....DataSource" error.

**Comment by Fjohn666** [comment]
are you sure it's these fonctions that triggered this error ? Maybe it's another one

**Comment by francisjervis** [comment]
Yes, I'm sure.

**Comment by ZhuchoG** [comment]
I tried to set this props to, error is here :(

**Comment by tomoueda** [comment]
Yeah I think the problem was misunderstood.

JSQMessagesViewController has properties: 
senderId
senderDisplayName

JSQMessagesCollectionViewDataSource has required functions named:

senderId() -> String
senderDisplayName() -> String

So when we try to do something like this:

extension customMessagesViewController: JSQMessagesCollectionViewDataSource {


}

**Comment by francisjervis** [comment]
Before the update I had the two functions returning the user's name/id, and then self.senderId and self.senderDisplayName set in viewDidLoad. Simply removing the functions means it doesn't conform to the datasource protocol.

**Comment by ge0rge** [comment]
Probably the quickest way to fix this is to refactor responsibilities of JSQMessagesCollectionViewDataSource out of custom subclass of JSQMessagesViewController and set datasource / delegate appropriately. Something along these lines:

```swift
class MyMessagesViewController: JSQMessagesViewController {
  // other methods
  // ...
  override func viewDidLoad() {
    super.viewDidLoad()
    // stuff..
    let messageDataSource = customDataSource: MyDataSource()
    collectionView.dataSource = messagesDataSource
  }
  // ...
  // more methods
}

class MyDataSource: JSQMessagesCollectionViewDataSource {
  func senderId() -> String { /** your implementation */ }
  func senderDisplayName() -> String { /** your implementation */ }
  // ...
  // other required methods
}
```

Let me know if this helps!

**Comment by ilandbt** [comment]
You cant have an attribute and a method have the same name or have a set/getAttributeName

So you need to change the names of the functions
for example:

...
- (NSString *)theSenderDisplayName;
- (NSString *)theSenderId;
...

And don't forget to change the names in

JSQMessagesCollectionViewFlowLayout.m

...
- (CGSize)jsq_avatarSizeForIndexPath:(NSIndexPath *)indexPath
{
    id<JSQMessageData> messageData = [self.collectionView.dataSource collectionView:self.collectionView messageDataForItemAtIndexPath:indexPath];
    NSString *messageSender = [messageData senderId];
   
    if ([messageSender isEqualToString:[self.collectionView.dataSource theSenderId]]) {
        return self.outgoingAvatarViewSize;
    }
    
    return self.incomingAvatarViewSize;
}
...

**Comment by steilerDev** [comment]
You could also create an observed variable: 

    var theSenderID: String {
      get {
        return self.senderId
      }
      set {
        senderId = newValue
      }
    }

but in general setting senderId in `viewDidLoad` should be enough.

**Comment by jessesquires** [comment]
Fixed in #1636
